hnn_core/network.py

```diff
@@ -825,7 +825,7 @@ def _create_drive_conns(self, target_populations, weights_by_receptor,
             for cellname in target_populations:
                 drive_conn['target_gids'].extend(self.gid_ranges[cellname])
                 drive_conn['target_type'] = cellname
-                drive_conn_by_cell[cellname] = drive_conn
+                drive_conn_by_cell[cellname] = drive_conn.copy()
```

Comment:
This is very insidious, I'm very surprised that the lines below this end up referencing the same object! But I can of course confirm that that's what happens.